example/models/food/v1/food_test.go

```diff
@@ -0,0 +1,259 @@
+	partTablesQuery := `SELECT count(t.tablename), t.tablename
+		FROM pg_tables t
+		LEFT JOIN pg_partitioned_table p ON p.partrelid = (SELECT oid FROM pg_class WHERE relname = t.tablename)
+		WHERE t.schemaname NOT IN ('pg_catalog', 'information_schema') AND p.partrelid IS NOT NULL AND t.tablename = $1
+		GROUP BY t.tablename;`
+
+	rows, err := pg.DB.Query(ctx, partTablesQuery, tableName)
+	require.NoErrorf(t, err, "TestSchemaFoodPasta: failed to count partitioned tables query: '\n%s\n'", partTablesQuery)
+	defer rows.Close()
+	var partTableCount int
+	var queryTableName string
+
+	for rows.Next() {
+		err = rows.Scan(&partTableCount, &queryTableName)
+		require.NoError(t, err)
+		fmt.Printf("%s %d\n", tableName, partTableCount)
+	}
+
+	require.NoError(t, rows.Err())
+	require.Equal(t, 1, partTableCount, "Should have one master partition table")
+	require.Equal(t, tableName, queryTableName, "Should have one master partition table")
+
+}
+
+func verifySubTables(t *testing.T, pg *pgtest.PG, tableName string, fakeTenantIds []string) {
+	ctx := context.Background()
+	// Verify number of sub tables
+	// Verify sub-partition tables
+	sqlSubTables := `SELECT
+			parent.relname      AS parent,
+			child.relname       AS child
+		FROM pg_inherits
+			JOIN pg_class parent            ON pg_inherits.inhparent = parent.oid
+			JOIN pg_class child             ON pg_inherits.inhrelid   = child.oid
+			JOIN pg_namespace nmsp_parent   ON nmsp_parent.oid  = parent.relnamespace
+			JOIN pg_namespace nmsp_child    ON nmsp_child.oid   = child.relnamespace
+		WHERE parent.relname='%s';`
+
+	rows, err := pg.DB.Query(ctx, fmt.Sprintf(sqlSubTables, tableName))
+	require.NoErrorf(t, err, "TestSchemaFoodPasta: failed to count partitioned tables query: '\n%s\n'", sqlSubTables)
+	defer rows.Close()
+
+	var parentTable string
+	var childTable string
+	rowCount := 0
+	selectedSubTableNames := make([]string, 0, len(fakeTenantIds))
+
+	for rows.Next() {
+		err = rows.Scan(&parentTable, &childTable)
+		require.NoError(t, err)
+		fmt.Printf("%s: %s\n", parentTable, childTable)
+		require.Equal(t, tableName, parentTable, "Parent table name did not match proto")
+		selectedSubTableNames = append(selectedSubTableNames, childTable)
+		rowCount += 1
+	}
+
+	require.NoError(t, rows.Err())
+	require.Equal(t, len(fakeTenantIds), rowCount, "Should have one sub-partition table per fake tenant")
+	for _, tenantId := range fakeTenantIds {
+		require.Contains(t, selectedSubTableNames, fmt.Sprintf("%s_%s", tableName, tenantId), "Should have a sub-partition table for tenant %s", tenantId)
+	}
+}
+
+func testInsertAndVerify(t *testing.T, pg *pgtest.PG, tableName string, fakeTenantIds []string, msg pgdb_v1.DBReflectMessage) {
+	ctx := context.Background()
+	// Insert data into master table
+	// Verify data in master table
+	// Verify data in sub tables
+	var obj1 pgdb_v1.DBReflectMessage
+	var obj2 pgdb_v1.DBReflectMessage
+	var obj3 pgdb_v1.DBReflectMessage
+	switch msg.(type) {
+	case *Pasta:
```

Comment:
maybe the test objects should be in the test case struct in the for loop?  aka like a "table test"

 feels weird for this to be in the inner func